Realm/RLMArray.mm

```diff
@@ -299,6 +299,13 @@ - (RLMResults *)objectsWhere:(NSString *)predicateFormat args:(va_list)args
     return [self objectsWithPredicate:[NSPredicate predicateWithFormat:predicateFormat arguments:args]];
 }
 
+- (id)valueForKeyPath:(NSString *)keyPath {
+    if (_backingArray) {
+        return [_backingArray valueForKeyPath:keyPath];
```

Comment:
`valueForKeyPath:` isn't overridden on `RLMArray` currently, but it does automatically forward `valueForKey:` to the backing array. Although if KVC operators only trigger `valueForKeyPath:` and not `valueForKey:` then you're right, you can't depend on this today.